src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -991,17 +1047,21 @@ public void TypeDeclaration_NoInstanceInitializers()
     public int Property { get; set; } = 43;
 }
 ";
-            // This will be allowed in C# 10, or we need to improve the message
-            // Tracking issue https://github.com/dotnet/roslyn/issues/52240
-            var comp = CreateCompilation(src);
+
+            var comp = CreateCompilation(src, parseOptions: TestOptions.Regular9);
             comp.VerifyDiagnostics(
-                // (4,16): error CS0573: 'S': cannot have instance property or field initializers in structs
+                // (2,15): error CS8652: The feature 'record structs' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // public record struct S
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "struct").WithArguments("record structs").WithLocation(2, 15),
+                // (4,16): error CS8652: The feature 'parameterless struct constructors' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
                 //     public int field = 42;
-                Diagnostic(ErrorCode.ERR_FieldInitializerInStruct, "field").WithArguments("S").WithLocation(4, 16),
-                // (5,16): error CS0573: 'S': cannot have instance property or field initializers in structs
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "field").WithArguments("parameterless struct constructors").WithLocation(4, 16),
+                // (5,16): error CS8652: The feature 'parameterless struct constructors' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
                 //     public int Property { get; set; } = 43;
-                Diagnostic(ErrorCode.ERR_FieldInitializerInStruct, "Property").WithArguments("S").WithLocation(5, 16)
-                );
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "Property").WithArguments("parameterless struct constructors").WithLocation(5, 16));
+
+            comp = CreateCompilation(src, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
```

Comment:
Please verify execution for this scenario, and also one with a primary constructor with and without parameters.